### PR TITLE
 Add warn for new versions

### PR DESCRIPTION
**Pull request by TsvetanMilanov**
When executing tns doctor outside of project if the nativescript cli needs update it will print information for the available update.
If tns doctor is executed in project it will check the installed runtimes versions and the nativescript core modules version and print information if an update is needed.
Add tns info command to display versions information for NativeScript CLI, core modules and runtimes.

### CODE REVIEW

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by TsvetanMilanov** [comment]
run ci

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by justcodebuilduser** [comment]
:heart:

File: declarations.d.ts [hunk 1]
```diff
@@ -722,9 +722,9 @@ interface IDoctorService {
 	/**
 	 * Verifies the host OS configuration and prints warnings to the users
 	 * @param configOptions: defines if the result should be tracked by Analytics
-	 * @returns {boolean} true if at least one warning was printed
+	 * @returns {IFuture<boolean>} true if at least one warning was printed
 	 */
-	printWarnings(configOptions?: { trackResult: boolean }): boolean;
+	printWarnings(configOptions?: { trackResult: boolean }): IFuture<boolean>;
```
Comment (rosen-vladimirov) [inline declarations.d.ts]: this method is used in several places across icenium-cli and nativescript-cli. Please make sure you have added `.wait()` everywhere or rollback the change.

File: declarations.d.ts [hunk 2]
```diff
@@ -1011,3 +1011,9 @@ interface ILiveSyncProvider {
 	 */
 	canExecuteFastSync(filePath: string, platform?: string): boolean;
 }
+
+interface IVersionInformation {
```
Comment (rosen-vladimirov) [inline declarations.d.ts]: please add JSDocs for the new interface

File: helpers.ts [hunk 3]
```diff
@@ -143,6 +143,33 @@ export function createTable(headers: string[], data: string[][]): any {
 	return table;
 }
 
+export function createTableWithVersionsInformation(versionsInformation: IVersionInformation[]): any {
```
Comment (rosen-vladimirov) [inline helpers.ts]: Is this method used on more than one place? Can we add it to a command/service or somewhere else as it seems quite specific. In `helpers` we add only very common methods that  can be used in various cases.

File: helpers.ts [hunk 4]
```diff
@@ -143,6 +143,33 @@ export function createTable(headers: string[], data: string[][]): any {
...
+	let headers = ["Component", "Current version", "Latest version", "Information"];
+	let data: string[][] = [[]];
+	let upToDate: string = "Up to date".green.toString();
+
+	_.forEach(versionsInformation, (componentInformation: IVersionInformation) => {
+		let row: string[] = [];
```
Comment (rosen-vladimirov) [inline helpers.ts]: Maybe:
```TypeScript
let row: string[] = [ componentInformation.componentName,
    componentInformation.currentVersion || "",
    componentInformation.latestVersion
];
```

File: helpers.ts [hunk 5]
```diff
@@ -143,6 +143,33 @@ export function createTable(headers: string[], data: string[][]): any {
...
+		row.push(componentInformation.componentName);
+		row.push(componentInformation.currentVersion || "");
+		row.push(componentInformation.latestVersion);
+
+		if (componentInformation.currentVersion) {
+			if (versionCompare(componentInformation.currentVersion, componentInformation.latestVersion) === -1) {
```
Comment (rosen-vladimirov) [inline helpers.ts]: can we use `semver` here instead of `versionCompare`

File: helpers.ts [hunk 6]
```diff
@@ -143,6 +143,33 @@ export function createTable(headers: string[], data: string[][]): any {
...
+	let data: string[][] = [[]];
+	let upToDate: string = "Up to date".green.toString();
+
+	_.forEach(versionsInformation, (componentInformation: IVersionInformation) => {
+		let row: string[] = [];
+		row.push(componentInformation.componentName);
+		row.push(componentInformation.currentVersion || "");
+		row.push(componentInformation.latestVersion);
+
+		if (componentInformation.currentVersion) {
```
Comment (rosen-vladimirov) [inline helpers.ts]: maybe we can improve this:
```TypeScript
let informationMsg = componentInformation.currentVersion && semver.lt(componentInformation.currentVersion, componentInformation.latestVersion) ? "Update available".yellow.toString() : upToDate;
row.push(informationMsg)
```

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by rosen-vladimirov** [comment]
:+1:  but please retarget the PR to release branch

**Comment by TsvetanMilanov** [comment]
Closing and retarget to release.

**Pull request closed without merging**